momento-provider/src/main/java/com/google/code/ssm/providers/momento/MomentoClientWrapper.java

```diff
@@ -253,18 +253,96 @@ private Object readFromMomento(final String key) {
                 .orElse(null);
     }
 
+    // Helper function that mimics multi-get by asynchronously calling get on all the keys. Large enough sets of keys
+    // are split into chunks to prevent too many simultaneous calls.
+    private <T> Map<String, T> readFromMomento(final Collection<String> keys, final CacheTranscoder transcoder) {
+        final Transcoder<T> cacheTranscoder = getTranscoder(transcoder);
+
+        final Set<Set<String>> keyChunks = splitKeyset(keys);
+
+        final Map<String, T> output = new HashMap<>();
+        for (Set<String> keyChunk : keyChunks) {
```

Comment:
This is fine for now.  The one shortcoming that it has is that we don't move on to the next batch until the slowest request in the first batch completes.  But I'm okay with that for the time being for sure.

If we revisit this in the future we should probably do something with a semaphore, where the semaphore has, say, 10 permits, and the loop just loops over all of the requests, tries to acquire a permit from the semaphore, then fires off the request.  And when any request completes we would add a permit back to the semaphore.  That way we're only bounded by the slowest request overall, instead of per-batch.  But that logic would be a little more complicated and require more testing and test coverage, so we don't need to do it now.  Just something to keep in mind for the future.